keycloak/keycloak-operator
# Add Pod Antiaffinity to keycloak deployment

PodAntiaffinity settings are added to keycloak deployment spec. When the
affinity values are provided in the CR it is used, else the default values are used.

## JIRA ID
https://issues.redhat.com/browse/KEYCLOAK-16482
https://issues.redhat.com/browse/MGDSTRM-1658


## Additional Information


Default podantiaffinity settings that would be used if none specified:
``` affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - weight: 100
        podAffinityTerm:
          labelSelector:
            matchExpressions:
            - key: app
              operator: In
              values:
              -keycloak
          topologyKey: "topology.kubernetes.io/zone"
      - weight: 90
        podAffinityTerm:
          labelSelector:
            matchExpressions:
            - key: app
              operator: In
              values:
              - keycloak
          topologyKey: "kubernetes.io/hostname"
```


## Verification Steps
> Unit tests are added to verify the deployment spec created
> deploy/examples/keycloak/keycloak-with-podantiaffinity.yaml - new example created that uses the podantiaffinity.

Add the steps required to check this change. Following an example.

1. Use the example as mentioned above or create a new spec with podantiaffinity specified
2. Deploy the keycloak CR (from the above step)
3. Verify that the keycloak pod is created and running successfully
4. Create a multi AZ cluster, deploy the keycloak and see if it works



## Checklist:
